web-ng/src/app/services/data-store/data-store.service.ts

```diff
@@ -74,18 +74,18 @@ export class DataStoreService {
       .then(() => projectId);
   }
 
-  updateProjectLayer(projectId: string, layers: string) {
-    const updatedLayers = JSON.parse(layers);
+  // TODO: Define return types for methods in this class
+  updateProjectLayer(projectId: string, layers: {}) {
```

Comment:
By accepting an `any` object with path:value pairs, this method exposes Firebase's API for updating data. DatastoreService abstracts away all Firebase-specific details (Separation of Concerns), so this method should really just take the following:
`updateProjectLayer(projectId: string, layerId: string, layer: Layer)`

And build the dot.notation.path itself instead of having the caller do it as above.